include/suggestion.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * A module to generate suggestions for a word
+ */
+
+#ifndef INCLUDE_SUGGESTION_H_
+#define INCLUDE_SUGGESTION_H_
+
+#include "trie.h"
+
+// The maximum length of a string we're willing to process
+// Longest word in english dictionary is 45 letters lol
+#define MAXLEN 60
+
+/* A simple way to store an approximate match and its score */
+typedef struct {
+    char *str;
+    int edits_left;
+} match_t;
+
+/*
+ * Sees if there are any words in a trie that contain a given prefix
+ * NOTE- awaiting the actual function from support tools
+ * 
+ * Parameters: 
+ *  - t: A trie. Must point to a trie allocated with trie_new
+ *  - s: A string. The prefix to check
+ * 
+ * Returns:
+ *  - EXIT_SUCCESS if there is a word in the dictionary with the prefix
+ *  - EXIT_FAILURE otherwise
+*/
+int has_children(trie_t *t, char *s);
```

Comment:
Let's have this return a `bool` instead.

Reply:
@dwahme

Reply:
has_children has been updated to return bools